File: bpsdka/bpsdka4130/Samples/Print-SDK-Demo-Android/AutoGen/BFElement/output/cpp/bf_element_set_print_settings_definition.cpp

#include "bf_element_set_print_settings_definition.hpp"
#include <string>
#include <cstring>

namespace bf {
  namespace element_set {
    const char* PrintSettings::convertToString(Number num) {
      switch(num) {
        case Number::MediaSize: return String::MediaSize;
        case Number::ImageOrientation: return String::ImageOrientation;
        case Number::PrintScalingMode: return String::PrintScalingMode;
        case Number::ImageHalftone: return String::ImageHalftone;
        case Number::ImageBrightness: return String::ImageBrightness;
        case Number::ImageContrast: return String::ImageContrast;
        case Number::ImageRedEnhancement: return String::ImageRedEnhancement;
        case Number::ImageGreenEnhancement: return String::ImageGreenEnhancement;
        case Number::ImageBlueEnhancement: return String::ImageBlueEnhancement;
        case Number::PrintHorizontalAlign: return String::PrintHorizontalAlign;
        case Number::PrintVerticalAlign: return String::PrintVerticalAlign;
        case Number::PrintTopMargin: return String::PrintTopMargin;
        case Number::PrintLeftMargin: return String::PrintLeftMargin;
        case Number::MediaTextColor: return String::MediaTextColor;
        case Number::MediaBackgroundColor: return String::MediaBackgroundColor;
        case Number::PrintScalingValue: return String::PrintScalingValue;
        case Number::PrintAutoCutLast: return String::PrintAutoCutLast;
        case Number::PrintAutoCutEvery: return String::PrintAutoCutEvery;
        case Number::PrintAutoCutEveryCyclePerPage: return String::PrintAutoCutEveryCyclePerPage;
        case Number::PrintAutoCutDisableForSpecialTape: return String::PrintAutoCutDisableForSpecialTape;
        case Number::PrintAutoHalfCut: return String::PrintAutoHalfCut;
        case Number::PrintDensity: return String::PrintDensity;
        case Number::PrintSpeed: return String::PrintSpeed;
        case Number::PrintDashLine: return String::PrintDashLine;
        case Number::PrintUsingCarbonCopyPaper: return String::PrintUsingCarbonCopyPaper;
        case Number::PrintFeedMode: return String::PrintFeedMode;
        case Number::PrintQuality: return String::PrintQuality;
        case Number::MediaPJRollCase: return String::MediaPJRollCase;
        case Number::MediaPaperInsertionPosition: return String::MediaPaperInsertionPosition;
        case Number::MediaPJPaperType: return String::MediaPJPaperType;
        case Number::MediaPJCustomSize: return String::MediaPJCustomSize;
        case Number::MediaPJForceStretchPrintableArea: return String::MediaPJForceStretchPrintableArea;
        case Number::PrintExtraFeedDots: return String::PrintExtraFeedDots;
        case Number::PrintCopies: return String::PrintCopies;
        case Number::PrintBidirectionalCommunication: return String::PrintBidirectionalCommunication;
        case Number::ImageTrimBlankDataFromTail: return String::ImageTrimBlankDataFromTail;
        case Number::PrintPolicyOfProcessingPrintData: return String::PrintPolicyOfProcessingPrintData;
        case Number::PrintCollate: return String::PrintCollate;
        case Number::PrintCheckMediaTextColor: return String::PrintCheckMediaTextColor;
        case Number::PrintCheckMediaBackgroundColor: return String::PrintCheckMediaBackgroundColor;
        case Number::PrintCutPause: return String::PrintCutPause;
        case Number::__Null: return String::__Null;
      }
      return String::__Null;
    };
    PrintSettings::Number PrintSettings::convertToNumber(const char* string) {
      if (!strcmp(string, String::MediaSize)) {
        return Number::MediaSize;
      }
      if (!strcmp(string, String::ImageOrientation)) {
        return Number::ImageOrientation;
      }
      if (!strcmp(string, String::PrintScalingMode)) {
        return Number::PrintScalingMode;
      }
      if (!strcmp(string, String::ImageHalftone)) {
        return Number::ImageHalftone;
      }
      if (!strcmp(string, String::ImageBrightness)) {
        return Number::ImageBrightness;
      }
      if (!strcmp(string, String::ImageContrast)) {
        return Number::ImageContrast;
      }
      if (!strcmp(string, String::ImageRedEnhancement)) {
        return Number::ImageRedEnhancement;
      }
      if (!strcmp(string, String::ImageGreenEnhancement)) {
        return Number::ImageGreenEnhancement;
      }
      if (!strcmp(string, String::ImageBlueEnhancement)) {
        return Number::ImageBlueEnhancement;
      }
      if (!strcmp(string, String::PrintHorizontalAlign)) {
        return Number::PrintHorizontalAlign;
      }
      if (!strcmp(string, String::PrintVerticalAlign)) {
        return Number::PrintVerticalAlign;
      }
      if (!strcmp(string, String::PrintTopMargin)) {
        return Number::PrintTopMargin;
      }
      if (!strcmp(string, String::PrintLeftMargin)) {
        return Number::PrintLeftMargin;
      }
      if (!strcmp(string, String::MediaTextColor)) {
        return Number::MediaTextColor;
      }
      if (!strcmp(string, String::MediaBackgroundColor)) {
        return Number::MediaBackgroundColor;
      }
      if (!strcmp(string, String::PrintScalingValue)) {
        return Number::PrintScalingValue;
      }
      if (!strcmp(string, String::PrintAutoCutLast)) {
        return Number::PrintAutoCutLast;
      }
      if (!strcmp(string, String::PrintAutoCutEvery)) {
        return Number::PrintAutoCutEvery;
      }
      if (!strcmp(string, String::PrintAutoCutEveryCyclePerPage)) {
        return Number::PrintAutoCutEveryCyclePerPage;
      }
      if (!strcmp(string, String::PrintAutoCutDisableForSpecialTape)) {
        return Number::PrintAutoCutDisableForSpecialTape;
      }
      if (!strcmp(string, String::PrintAutoHalfCut)) {
        return Number::PrintAutoHalfCut;
      }
      if (!strcmp(string, String::PrintDensity)) {
        return Number::PrintDensity;
      }
      if (!strcmp(string, String::PrintSpeed)) {
        return Number::PrintSpeed;
      }
      if (!strcmp(string, String::PrintDashLine)) {
        return Number::PrintDashLine;
      }
      if (!strcmp(string, String::PrintUsingCarbonCopyPaper)) {
        return Number::PrintUsingCarbonCopyPaper;
      }
      if (!strcmp(string, String::PrintFeedMode)) {
        return Number::PrintFeedMode;
      }
      if (!strcmp(string, String::PrintQuality)) {
        return Number::PrintQuality;
      }
      if (!strcmp(string, String::MediaPJRollCase)) {
        return Number::MediaPJRollCase;
      }
      if (!strcmp(string, String::MediaPaperInsertionPosition)) {
        return Number::MediaPaperInsertionPosition;
      }
      if (!strcmp(string, String::MediaPJPaperType)) {
        return Number::MediaPJPaperType;
      }
      if (!strcmp(string, String::MediaPJCustomSize)) {
        return Number::MediaPJCustomSize;
      }
      if (!strcmp(string, String::MediaPJForceStretchPrintableArea)) {
        return Number::MediaPJForceStretchPrintableArea;
      }
      if (!strcmp(string, String::PrintExtraFeedDots)) {
        return Number::PrintExtraFeedDots;
      }
      if (!strcmp(string, String::PrintCopies)) {
        return Number::PrintCopies;
      }
      if (!strcmp(string, String::PrintBidirectionalCommunication)) {
        return Number::PrintBidirectionalCommunication;
      }
      if (!strcmp(string, String::ImageTrimBlankDataFromTail)) {
        return Number::ImageTrimBlankDataFromTail;
      }
      if (!strcmp(string, String::PrintPolicyOfProcessingPrintData)) {
        return Number::PrintPolicyOfProcessingPrintData;
      }
      if (!strcmp(string, String::PrintCollate)) {
        return Number::PrintCollate;
      }
      if (!strcmp(string, String::PrintCheckMediaTextColor)) {
        return Number::PrintCheckMediaTextColor;
      }
      if (!strcmp(string, String::PrintCheckMediaBackgroundColor)) {
        return Number::PrintCheckMediaBackgroundColor;
      }
      if (!strcmp(string, String::PrintCutPause)) {
        return Number::PrintCutPause;
      }
      if (!strcmp(string, String::__Null)) {
        return Number::__Null;
      }
      return Number::__Null;
    };

  }
}

File: bpsdka/bpsdka4130/Samples/Print-SDK-Demo-Android/AutoGen/BFElement/output/oc/BFElementSetPrintSettingsDefinition.m


#import "BFElementSetPrintSettingsDefinition.h"

NSString * __nonnull const BFElementSetPrintSettingsStringMediaSize = @"MediaSize";
NSString * __nonnull const BFElementSetPrintSettingsStringImageOrientation = @"ImageOrientation";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintScalingMode = @"PrintScalingMode";
NSString * __nonnull const BFElementSetPrintSettingsStringImageHalftone = @"ImageHalftone";
NSString * __nonnull const BFElementSetPrintSettingsStringImageBrightness = @"ImageBrightness";
NSString * __nonnull const BFElementSetPrintSettingsStringImageContrast = @"ImageContrast";
NSString * __nonnull const BFElementSetPrintSettingsStringImageRedEnhancement = @"ImageRedEnhancement";
NSString * __nonnull const BFElementSetPrintSettingsStringImageGreenEnhancement = @"ImageGreenEnhancement";
NSString * __nonnull const BFElementSetPrintSettingsStringImageBlueEnhancement = @"ImageBlueEnhancement";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintHorizontalAlign = @"PrintHorizontalAlign";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintVerticalAlign = @"PrintVerticalAlign";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintTopMargin = @"PrintTopMargin";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintLeftMargin = @"PrintLeftMargin";
NSString * __nonnull const BFElementSetPrintSettingsStringMediaTextColor = @"MediaTextColor";
NSString * __nonnull const BFElementSetPrintSettingsStringMediaBackgroundColor = @"MediaBackgroundColor";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintScalingValue = @"PrintScalingValue";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintAutoCutLast = @"PrintAutoCutLast";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintAutoCutEvery = @"PrintAutoCutEvery";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintAutoCutEveryCyclePerPage = @"PrintAutoCutEveryCyclePerPage";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintAutoCutDisableForSpecialTape = @"PrintAutoCutDisableForSpecialTape";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintAutoHalfCut = @"PrintAutoHalfCut";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintDensity = @"PrintDensity";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintSpeed = @"PrintSpeed";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintDashLine = @"PrintDashLine";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintUsingCarbonCopyPaper = @"PrintUsingCarbonCopyPaper";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintFeedMode = @"PrintFeedMode";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintQuality = @"PrintQuality";
NSString * __nonnull const BFElementSetPrintSettingsStringMediaPJRollCase = @"MediaPJRollCase";
NSString * __nonnull const BFElementSetPrintSettingsStringMediaPaperInsertionPosition = @"MediaPaperInsertionPosition";
NSString * __nonnull const BFElementSetPrintSettingsStringMediaPJPaperType = @"MediaPJPaperType";
NSString * __nonnull const BFElementSetPrintSettingsStringMediaPJCustomSize = @"MediaPJCustomSize";
NSString * __nonnull const BFElementSetPrintSettingsStringMediaPJForceStretchPrintableArea = @"MediaPJForceStretchPrintableArea";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintExtraFeedDots = @"PrintExtraFeedDots";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintCopies = @"PrintCopies";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintBidirectionalCommunication = @"PrintBidirectionalCommunication";
NSString * __nonnull const BFElementSetPrintSettingsStringImageTrimBlankDataFromTail = @"ImageTrimBlankDataFromTail";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintPolicyOfProcessingPrintData = @"PrintPolicyOfProcessingPrintData";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintCollate = @"PrintCollate";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintCheckMediaTextColor = @"PrintCheckMediaTextColor";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintCheckMediaBackgroundColor = @"PrintCheckMediaBackgroundColor";
NSString * __nonnull const BFElementSetPrintSettingsStringPrintCutPause = @"PrintCutPause";
NSString * __nonnull const BFElementSetPrintSettingsString__Null = @"__Null";
@implementation BFElementSetPrintSettings
+ (nonnull NSString*)convertToString:(BFElementSetPrintSettingsNumber)num {
  switch(num) {
  case BFElementSetPrintSettingsNumberMediaSize: return BFElementSetPrintSettingsStringMediaSize;
  case BFElementSetPrintSettingsNumberImageOrientation: return BFElementSetPrintSettingsStringImageOrientation;
  case BFElementSetPrintSettingsNumberPrintScalingMode: return BFElementSetPrintSettingsStringPrintScalingMode;
  case BFElementSetPrintSettingsNumberImageHalftone: return BFElementSetPrintSettingsStringImageHalftone;
  case BFElementSetPrintSettingsNumberImageBrightness: return BFElementSetPrintSettingsStringImageBrightness;
  case BFElementSetPrintSettingsNumberImageContrast: return BFElementSetPrintSettingsStringImageContrast;
  case BFElementSetPrintSettingsNumberImageRedEnhancement: return BFElementSetPrintSettingsStringImageRedEnhancement;
  case BFElementSetPrintSettingsNumberImageGreenEnhancement: return BFElementSetPrintSettingsStringImageGreenEnhancement;
  case BFElementSetPrintSettingsNumberImageBlueEnhancement: return BFElementSetPrintSettingsStringImageBlueEnhancement;
  case BFElementSetPrintSettingsNumberPrintHorizontalAlign: return BFElementSetPrintSettingsStringPrintHorizontalAlign;
  case BFElementSetPrintSettingsNumberPrintVerticalAlign: return BFElementSetPrintSettingsStringPrintVerticalAlign;
  case BFElementSetPrintSettingsNumberPrintTopMargin: return BFElementSetPrintSettingsStringPrintTopMargin;
  case BFElementSetPrintSettingsNumberPrintLeftMargin: return BFElementSetPrintSettingsStringPrintLeftMargin;
  case BFElementSetPrintSettingsNumberMediaTextColor: return BFElementSetPrintSettingsStringMediaTextColor;
  case BFElementSetPrintSettingsNumberMediaBackgroundColor: return BFElementSetPrintSettingsStringMediaBackgroundColor;
  case BFElementSetPrintSettingsNumberPrintScalingValue: return BFElementSetPrintSettingsStringPrintScalingValue;
  case BFElementSetPrintSettingsNumberPrintAutoCutLast: return BFElementSetPrintSettingsStringPrintAutoCutLast;
  case BFElementSetPrintSettingsNumberPrintAutoCutEvery: return BFElementSetPrintSettingsStringPrintAutoCutEvery;
  case BFElementSetPrintSettingsNumberPrintAutoCutEveryCyclePerPage: return BFElementSetPrintSettingsStringPrintAutoCutEveryCyclePerPage;
  case BFElementSetPrintSettingsNumberPrintAutoCutDisableForSpecialTape: return BFElementSetPrintSettingsStringPrintAutoCutDisableForSpecialTape;
  case BFElementSetPrintSettingsNumberPrintAutoHalfCut: return BFElementSetPrintSettingsStringPrintAutoHalfCut;
  case BFElementSetPrintSettingsNumberPrintDensity: return BFElementSetPrintSettingsStringPrintDensity;
  case BFElementSetPrintSettingsNumberPrintSpeed: return BFElementSetPrintSettingsStringPrintSpeed;
  case BFElementSetPrintSettingsNumberPrintDashLine: return BFElementSetPrintSettingsStringPrintDashLine;
  case BFElementSetPrintSettingsNumberPrintUsingCarbonCopyPaper: return BFElementSetPrintSettingsStringPrintUsingCarbonCopyPaper;
  case BFElementSetPrintSettingsNumberPrintFeedMode: return BFElementSetPrintSettingsStringPrintFeedMode;
  case BFElementSetPrintSettingsNumberPrintQuality: return BFElementSetPrintSettingsStringPrintQuality;
  case BFElementSetPrintSettingsNumberMediaPJRollCase: return BFElementSetPrintSettingsStringMediaPJRollCase;
  case BFElementSetPrintSettingsNumberMediaPaperInsertionPosition: return BFElementSetPrintSettingsStringMediaPaperInsertionPosition;
  case BFElementSetPrintSettingsNumberMediaPJPaperType: return BFElementSetPrintSettingsStringMediaPJPaperType;
  case BFElementSetPrintSettingsNumberMediaPJCustomSize: return BFElementSetPrintSettingsStringMediaPJCustomSize;
  case BFElementSetPrintSettingsNumberMediaPJForceStretchPrintableArea: return BFElementSetPrintSettingsStringMediaPJForceStretchPrintableArea;
  case BFElementSetPrintSettingsNumberPrintExtraFeedDots: return BFElementSetPrintSettingsStringPrintExtraFeedDots;
  case BFElementSetPrintSettingsNumberPrintCopies: return BFElementSetPrintSettingsStringPrintCopies;
  case BFElementSetPrintSettingsNumberPrintBidirectionalCommunication: return BFElementSetPrintSettingsStringPrintBidirectionalCommunication;
  case BFElementSetPrintSettingsNumberImageTrimBlankDataFromTail: return BFElementSetPrintSettingsStringImageTrimBlankDataFromTail;
  case BFElementSetPrintSettingsNumberPrintPolicyOfProcessingPrintData: return BFElementSetPrintSettingsStringPrintPolicyOfProcessingPrintData;
  case BFElementSetPrintSettingsNumberPrintCollate: return BFElementSetPrintSettingsStringPrintCollate;
  case BFElementSetPrintSettingsNumberPrintCheckMediaTextColor: return BFElementSetPrintSettingsStringPrintCheckMediaTextColor;
  case BFElementSetPrintSettingsNumberPrintCheckMediaBackgroundColor: return BFElementSetPrintSettingsStringPrintCheckMediaBackgroundColor;
  case BFElementSetPrintSettingsNumberPrintCutPause: return BFElementSetPrintSettingsStringPrintCutPause;
  case BFElementSetPrintSettingsNumber__Null: return BFElementSetPrintSettingsString__Null;
  }
  return BFElementSetPrintSettingsString__Null;
}
+ (BFElementSetPrintSettingsNumber)convertToNumber:(nonnull NSString*)str {
  if ([str isEqualToString:BFElementSetPrintSettingsStringMediaSize]) {
    return BFElementSetPrintSettingsNumberMediaSize;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringImageOrientation]) {
    return BFElementSetPrintSettingsNumberImageOrientation;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintScalingMode]) {
    return BFElementSetPrintSettingsNumberPrintScalingMode;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringImageHalftone]) {
    return BFElementSetPrintSettingsNumberImageHalftone;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringImageBrightness]) {
    return BFElementSetPrintSettingsNumberImageBrightness;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringImageContrast]) {
    return BFElementSetPrintSettingsNumberImageContrast;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringImageRedEnhancement]) {
    return BFElementSetPrintSettingsNumberImageRedEnhancement;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringImageGreenEnhancement]) {
    return BFElementSetPrintSettingsNumberImageGreenEnhancement;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringImageBlueEnhancement]) {
    return BFElementSetPrintSettingsNumberImageBlueEnhancement;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintHorizontalAlign]) {
    return BFElementSetPrintSettingsNumberPrintHorizontalAlign;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintVerticalAlign]) {
    return BFElementSetPrintSettingsNumberPrintVerticalAlign;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintTopMargin]) {
    return BFElementSetPrintSettingsNumberPrintTopMargin;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintLeftMargin]) {
    return BFElementSetPrintSettingsNumberPrintLeftMargin;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringMediaTextColor]) {
    return BFElementSetPrintSettingsNumberMediaTextColor;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringMediaBackgroundColor]) {
    return BFElementSetPrintSettingsNumberMediaBackgroundColor;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintScalingValue]) {
    return BFElementSetPrintSettingsNumberPrintScalingValue;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintAutoCutLast]) {
    return BFElementSetPrintSettingsNumberPrintAutoCutLast;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintAutoCutEvery]) {
    return BFElementSetPrintSettingsNumberPrintAutoCutEvery;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintAutoCutEveryCyclePerPage]) {
    return BFElementSetPrintSettingsNumberPrintAutoCutEveryCyclePerPage;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintAutoCutDisableForSpecialTape]) {
    return BFElementSetPrintSettingsNumberPrintAutoCutDisableForSpecialTape;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintAutoHalfCut]) {
    return BFElementSetPrintSettingsNumberPrintAutoHalfCut;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintDensity]) {
    return BFElementSetPrintSettingsNumberPrintDensity;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintSpeed]) {
    return BFElementSetPrintSettingsNumberPrintSpeed;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintDashLine]) {
    return BFElementSetPrintSettingsNumberPrintDashLine;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintUsingCarbonCopyPaper]) {
    return BFElementSetPrintSettingsNumberPrintUsingCarbonCopyPaper;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintFeedMode]) {
    return BFElementSetPrintSettingsNumberPrintFeedMode;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintQuality]) {
    return BFElementSetPrintSettingsNumberPrintQuality;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringMediaPJRollCase]) {
    return BFElementSetPrintSettingsNumberMediaPJRollCase;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringMediaPaperInsertionPosition]) {
    return BFElementSetPrintSettingsNumberMediaPaperInsertionPosition;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringMediaPJPaperType]) {
    return BFElementSetPrintSettingsNumberMediaPJPaperType;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringMediaPJCustomSize]) {
    return BFElementSetPrintSettingsNumberMediaPJCustomSize;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringMediaPJForceStretchPrintableArea]) {
    return BFElementSetPrintSettingsNumberMediaPJForceStretchPrintableArea;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintExtraFeedDots]) {
    return BFElementSetPrintSettingsNumberPrintExtraFeedDots;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintCopies]) {
    return BFElementSetPrintSettingsNumberPrintCopies;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintBidirectionalCommunication]) {
    return BFElementSetPrintSettingsNumberPrintBidirectionalCommunication;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringImageTrimBlankDataFromTail]) {
    return BFElementSetPrintSettingsNumberImageTrimBlankDataFromTail;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintPolicyOfProcessingPrintData]) {
    return BFElementSetPrintSettingsNumberPrintPolicyOfProcessingPrintData;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintCollate]) {
    return BFElementSetPrintSettingsNumberPrintCollate;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintCheckMediaTextColor]) {
    return BFElementSetPrintSettingsNumberPrintCheckMediaTextColor;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintCheckMediaBackgroundColor]) {
    return BFElementSetPrintSettingsNumberPrintCheckMediaBackgroundColor;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsStringPrintCutPause]) {
    return BFElementSetPrintSettingsNumberPrintCutPause;
  }
  if ([str isEqualToString:BFElementSetPrintSettingsString__Null]) {
    return BFElementSetPrintSettingsNumber__Null;
  }
  return BFElementSetPrintSettingsNumber__Null;
}
+ (nonnull NSString*)key {
  return @"PrintSettings";
}
@end

